### ISSUE TITLE
[0.40.0] experimental:spacing-between-declarations-with-comments false positives with `//` comments on class properties, variables and enum values

### ISSUE THREAD

**Issue by mateuszkwiecinski** [pinterest/ktlint#1053]
## Expected Behavior
I expect ktlint not to fail nor modify the code below when autoformatting. According to the issue [linked in ktlint check](https://github.com/pinterest/ktlint/blob/09e27487ed3dd1c0c568da4796e30f744f71c809/ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/SpacingBetweenDeclarationsWithCommentsRule.kt#L16), while Intellj indeed adds spacing between methods it doesn't between class properties, enum values, nor method variables.
Basically I expect `spacing-between-declarations-with-comments` not to report issues with a code that doesn't get autocorrected by IntelliJ.

## Observed Behavior
ktlint fails with 
```
SampleClass.kt:6:36: Declarations and declarations with comments should have an empty space between.
```
on every `"// false positive"` comment (below)

\+ when executed with `-F` it adds spaces in places where intellij doesn't
|before|after formatting with ktlint|
|---|---|
|![image](https://user-images.githubusercontent.com/36954793/104193977-020d1d80-5421-11eb-8869-a536609f2db6.png)|![image](https://user-images.githubusercontent.com/36954793/104193999-09342b80-5421-11eb-8e84-d50d9a955546.png)|

## Steps to Reproduce
Run ktlint on following code:
```kotlin
package com.doesnt.work.library

class SampleClass {

    private val public = "ok" // ok
    private val private = "not_ok" // false positive
    private val halfPublicHalfPrivate = "not_ok" // false positive

    /**
     * Doc 1
     */
    fun one() = 1

    /**
     * Doc 2
     */
    fun two() = 2
    fun three() {
        val public = "ok" // ok
        val private = "not_ok" // false positive
    }
}

enum class SampleEnum {
    One, // ok
    Two, // false positive
    Three, // false positive
}
```

## Your Environment
```groovy
configurations {
    ktlint
}

dependencies {
    ktlint "com.pinterest:ktlint:0.40.0"
}

task ktlint(type: JavaExec, group: "verification") {
    description = "Check Kotlin code style."
    classpath = configurations.ktlint
    main = "com.pinterest.ktlint.Main"
    args "src/**/*.kt", "--experimental"
}
```

* Version of ktlint used: 0.39.0 (works), 0.40.0 (doesn't work)
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): manually added Gradle tasks pasted above
* Version of Gradle used (if applicable): 6.8
* Operating System and version: 
* Link to your project (if it's a public repository):
